src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3196,56 +3330,142 @@ public virtual bool CoercesTo(DType typeDest, out bool isSafe, out DType coercio
                     // String to DateTime isn't safe for ill-formatted strings.
                     isSafe = Kind != DKind.String;
                     doesCoerce = Kind == DKind.String ||
-                                 Number.Accepts(this) ||
-                                 Decimal.Accepts(this) ||
-                                 DateTime.Accepts(this);
+                                 Number.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 Decimal.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 DateTime.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 Time.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules) ||
+                                 Date.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules);
```

Comment:
Are there just gaps in existing implementation?

Reply:
Before DateTime would accept Time and Date, so those would be covered by that. Now they aren't accepted anymore, so we need to explicitly add allow coercion here.